paasta_tools/setup_istio_mesh.py

```diff
@@ -0,0 +1,235 @@
+    pass
+
+
+class ErrorGettingServiceList(Exception):
+    pass
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Creates Kubernetes services.")
+    parser.add_argument(
+        "service_instance_list",
+        nargs="+",
+        help="The list of Kubernetes service instances to create or update services for",
+        metavar="SERVICE%sINSTANCE" % SPACER,
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", dest="verbose", default=False,
+    )
+
+    parser.add_argument(
+        "--dry-run", action="store_true", dest="dry_run", default=False,
+    )
+
+    parser.add_argument(
+        "-l",
+        "--rate-limit",
+        dest="rate_limit",
+        default=0,
+        metavar="LIMIT",
+        type=int,
+        help="Update or create up to this number of service instances. Default is 0 (no limit).",
+    )
+    parser.add_argument(
+        "-f",
+        "--file-path",
+        dest="file_path",
+        default="/nail/etc/services/services.json",
+        metavar="LIMIT",
+        type=str,
+        help="services file to use for smartstack namespace. Default file is /nail/etc/services/services.json",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def build_namespace_port_mapping(file_path: str) -> Mapping:
+    paasta_namespaces = {}
+    with open(file_path) as service_file:
+        smartstack_namespaces = json.load(service_file)
+        for namespace, endpoint in smartstack_namespaces.items():
+            paasta_namespaces[namespace] = endpoint["port"]
+    return paasta_namespaces
+
+
+def format_service_tuple(services: k8s.V1ServiceList) -> Sequence[KubernetesService]:
+    return [
+        KubernetesService(name=item.metadata.name, labels=item.metadata.labels)
+        for item in services.items
+    ]
+
+
+def sanitise_kubernetes_service_name(name: str) -> str:
+    name = sanitise_kubernetes_name(name)
+    return name.replace(".", "--")
+
+
+def get_existing_kubernetes_service_names(kube_client: KubeClient) -> Sequence:
+    service_objects = kube_client.core.list_namespaced_service("paasta")
+    if not service_objects:
+        raise ErrorGettingServiceList("Error retrieving services list from k8s api")
+
+    existing_kube_services = [
+        KubernetesService(name=item.metadata.name, labels=item.metadata.labels)
+        for item in service_objects.items
+    ]
+    return {item.name for item in existing_kube_services}
+
+
+def setup_unified_service(
+    kube_client: KubeClient, port_list: Sequence
+) -> k8s.V1Service:
+    service_meta = k8s.V1ObjectMeta(name=UNIFIED_K8S_SVC_NAME)
+
+    # Add port 1337 for envoy unified listener.
+    # Clients can connect to this listenner and set x-yelp-svc header for routing
+    ports = [
+        k8s.V1ServicePort(
+            name=f"p{UNIFIED_SVC_PORT}",
+            port=UNIFIED_SVC_PORT,
+            protocol="TCP",
+            target_port=PAASTA_SVC_PORT,
```

Comment:
we need appProtocol=http, otherwise istio is trying to guess protocol from service name